fsw/cfe-core/src/inc/cfe_sb.h

```diff
@@ -513,7 +514,40 @@ CFE_Status_t CFE_SB_UnsubscribeLocal(CFE_SB_MsgId_t MsgId, CFE_SB_PipeId_t PipeI
 
 /*****************************************************************************/
 /**
-** \brief Send a software bus message
+** \brief Transmit a message
+**
+** \par Description
+**          This routine copies the specified message into a software bus
+**          buffer which is then transmitted to all subscribers.  The
+**          software bus will read the message ID from the message header to
+**          determine which pipes should receive the message.
+**
+** \par Assumptions, External Events, and Notes:
+**          - This routine will not normally wait for the receiver tasks to
+**            process the message before returning control to the caller's task.
+**          - However, if a higher priority task is pending and subscribed to
+**            this message, that task may get to run before returning
+**            control to the caller.
+**
+** \param[in]  MsgPtr       A pointer to the message to be sent.  This must point
+**                          to the first byte of the message header.
+** \param[in] IncrementSequenceCount Boolean to increment the internally tracked
+**                                   sequence count and update the message if the
+**                                   buffer contains a telemetry message
+**
+** \return Execution status, see \ref CFEReturnCodes
+** \retval #CFE_SUCCESS         \copybrief CFE_SUCCESS
+** \retval #CFE_SB_BAD_ARGUMENT \copybrief CFE_SB_BAD_ARGUMENT
+** \retval #CFE_SB_MSG_TOO_BIG  \copybrief CFE_SB_MSG_TOO_BIG
+** \retval #CFE_SB_BUF_ALOC_ERR \copybrief CFE_SB_BUF_ALOC_ERR
+**/
+CFE_Status_t  CFE_SB_TransmitMsg(CFE_MSG_Message_t *MsgPtr, bool IncrementSequenceCount);
```

Comment:
I'm slightly concerned about the extra boolean on this API.  I would prefer something that can be a drop-in replacement for `CFE_SB_SendMsg`.  Given that probably 99% of users will be passing `true` for the sequence count, and passing `false` is probably only a special use case that only SBN needs.  It might be confusing to users why this extra `true` argument needs to be passed and open up the possibility that they make mistakes in doing so, then wonder why their sequence numbers are wrong.

For consideration - can we make the "increment sequence count" to be a property of the _buffer itself_ instead of the API call?  That is, the buffer indicates if it is a truly a "new" message or if it is simply re-encapsulating a message that originated elsewhere - and therefore doesn't get a new sequence#.

This means that `CFE_SB_TransmitMessage` would always increment sequence (no extra boolean parameter)

For the SBN use case, it could request a pass-through buffer and use the `CFE_SB_TransmitBuffer` to pass a message without incrementing the sequence count.

To put it simply - I think we should strive to keep the common use case (TransmitMsg) to be as simple of an API as possible, minimizing the number of ways they can go wrong.

Users who need something more complex or involving special treatment should use the buffer-based/zero copy pattern (i.e. get a buffer, indicate that the _buffer_ represents a pass-through message, and then send the buffer).

Users who are simply sending a new message from their app (most common use case) should get the simplest API which does exactly what they want and nothing else.